core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -130,12 +136,47 @@ public static UnderFileSystemFactory find(String path) {
       }
     }
 
+    // Scan extensions directory on the fly
+    eligibleFactories.addAll(findAllExtensions(path));
```

Comment:
this is expensive and could have a negative impact on performance if we need to do it repeatedly. Do you think it makes sense to instead only scan at startup? Alternately, we could do an asynchronous scan every 30 seconds or so